modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -4740,6 +4740,12 @@ GDScriptParser::DataType GDScriptAnalyzer::get_operation_type(Variant::Operator
 			b_type = Variant::INT;
 		}
 	}
+	if (p_operation == Variant::OP_ADD && a_type == Variant::ARRAY && b_type == Variant::ARRAY) {
+		if (p_a.get_container_element_type() == p_a.get_container_element_type()) {
+			r_valid = true;
+			return p_a;
```

Comment:
Small thing, `type_source` is an important part, let's say that `p_a` is a hard annotated type while `p_b` is a weak inferred one. Result should be a weak one.

So makes sense to move next lines of `bool hard_operation = p_a.is_hard_type() && p_b.is_hard_type();` and `GDScriptParser::DataType result;` before this if condition and have this inside condition:
```cpp
result = p_a;
result.type_source = hard_operation ? GDScriptParser::DataType::ANNOTATED_INFERRED : GDScriptParser::DataType::INFERRED;
```